service/src/main/java/org/broadinstitute/listener/relay/inspectors/SamResourceClient.java

```diff
@@ -41,6 +41,13 @@ public Instant checkPermission(String accessToken) {
         samClient.setAccessToken(accessToken);
         var resourceApi = new ResourcesApi(samClient);
 
+        // check that user has access to workspace
+        var workspaceAccess = resourceApi.resourcePermissionV2("workspace", samResourceId, "read");
```

Comment:
I think `samResourceId` is wrong -- that would be the ID of the runtime/kubernetes-app, not the workspace.

You might need to add a new contructor-arg to this class and pass it from `AppConfiguration`